adapters/sdk.go

```diff
@@ -208,11 +208,33 @@ func (b *SDKAdapter) GetBlockByID(
 			Height:    flowBlock.Header.Height,
 			Timestamp: flowBlock.Header.Timestamp,
 		},
-		BlockPayload: sdk.BlockPayload{},
+		BlockPayload: convertBlockPayload(flowBlock.Payload),
 	}
 	return &block, sdk.BlockStatusSealed, nil
 }
 
+func convertBlockPayload(payload *flowgo.Payload) sdk.BlockPayload {
+	seals := make([]*sdk.BlockSeal, 0)
+	for _, seal := range payload.Seals {
+		seals = append(seals, &sdk.BlockSeal{
+			BlockID:            sdk.Identifier(seal.BlockID),
+			ExecutionReceiptID: sdk.Identifier(seal.ResultID),
+		})
+	}
+
+	collectionGuarantees := make([]*sdk.CollectionGuarantee, 0)
+	for _, guarantee := range payload.Guarantees {
+		collectionGuarantees = append(collectionGuarantees, &sdk.CollectionGuarantee{
+			CollectionID: sdk.Identifier(guarantee.CollectionID),
+		})
+	}
```

Comment:
Pre-allocate, if needed:

```suggestion
	var seals []*sdk.BlockSeal
	sealCount := len(payload.Seals)
	if sealCount > 0 {
		seals = make([]*sdk.BlockSeal, 0, sealCount)
		for _, seal := range payload.Seals {
			seals = append(seals, &sdk.BlockSeal{
				BlockID:            sdk.Identifier(seal.BlockID),
				ExecutionReceiptID: sdk.Identifier(seal.ResultID),
			})
		}
	}

	var collectionGuarantees []*sdk.CollectionGuarantee
	guaranteesCount := len(payload.Guarantees)
	if guaranteesCount > 0 {
		collectionGuarantees = make([]*sdk.CollectionGuarantee, 0, guaranteesCount)
		for _, guarantee := range payload.Guarantees {
			collectionGuarantees = append(collectionGuarantees, &sdk.CollectionGuarantee{
				CollectionID: sdk.Identifier(guarantee.CollectionID),
			})
		}
	}
```

Reply:
Fair point, fixed in https://github.com/onflow/flow-emulator/pull/704/commits/5fc70a20f052256cad4476710dc0ff6093c39a4a